src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+            return true;
+        }
+        Node<T> preLast = last;
+        preLast.next = new Node<>(preLast, value, null);
+        last = preLast.next;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if ((isEmpty() && index == 0) || index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> add = new Node<>(null, value, first);
+            first.prev = add;
+            first = add;
+            size++;
+        } else {
+            validateIndex(index);
+            Node<T> node = getNode(index);
+            Node<T> add = new Node<>(node.prev, value, node);
+            node.prev.next = add;
+            node.prev = add;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
```

Comment:
let's make the variable name more informative

Reply:
done